tests/mocks/MockERC721.cairo

```diff
@@ -0,0 +1,368 @@
+    assert_not_zero(recipient)
+
+    let (owner_balance) = _balances.read(sender)
+    let (new_owner_balance: Uint256) = uint256_sub(owner_balance, Uint256(0,1))
+
+    let (recipient_balance) = _balances.read(recipient)
+    let (new_recipient_balance, _: Uint256) = uint256_add(recipient_balance, Uint256(0,1))
+
+    _balances.write(sender, new_owner_balance)
+    _balances.write(recipient, new_recipient_balance)
+
+    _owners.write(token_id, recipient)
+
+    _token_approvals.write(token_id, 0)
+
+    return ()
+end
+
+@external
+func transfer_from{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    bitwise_ptr : BitwiseBuiltin*,
+    range_check_ptr
+}(
+    sender: felt,
+    recipient: felt,
+    token_id: Uint256
+):
+    let (caller) = get_caller_address()
+    let (owner) = _owners.read(token_id)
+
+    assert sender = owner # wrong sender
+
+    assert_not_zero(recipient)
+
+    if caller == owner:
+        tempvar caller_is_owner = 1
+    else:
+        tempvar caller_is_owner = 0
+    end
+
+    let (approved_spender) = _token_approvals.read(token_id)
+    
+    if approved_spender == caller:
+        tempvar is_approved = 1
+    else:
+        tempvar is_approved = 0
+    end
+
+    let (is_approved_for_all) = _is_approved_for_all.read(owner, caller)
+    let (can_transfer1) = bitwise_or(caller_is_owner, is_approved)
+    let (can_transfer) = bitwise_or(can_transfer1, is_approved_for_all)
+    assert can_transfer = 1
+
+    let (owner_balance) = _balances.read(sender)
+    let (new_owner_balance: Uint256) = uint256_sub(owner_balance, Uint256(0,1))
+
+    let (recipient_balance) = _balances.read(recipient)
+    let (new_recipient_balance, _: Uint256) = uint256_add(recipient_balance, Uint256(0,1))
+
+    _balances.write(sender, new_owner_balance)
+    _balances.write(recipient, new_recipient_balance)
+
+    _owners.write(token_id, recipient)
+
+    _token_approvals.write(token_id, 0)
+
+    return ()
+end
+
+@external
+func mint{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    bitwise_ptr : BitwiseBuiltin*,
+    range_check_ptr
+}(
+    recipient: felt,
+    token_id: Uint256
+):
+    _mint(recipient, token_id)
+    return ()
+end
+
+@external
+func burn{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    bitwise_ptr : BitwiseBuiltin*,
+    range_check_ptr
+}(
+    token_id: Uint256
+):
+    _burn(token_id)
+    return ()
+end
+
+#############################################
+##             INTERNAL LOGIC              ##
+#############################################
+
+func _mint{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}(
+    recipient: felt,
+    token_id: Uint256
+):
+    assert_not_zero(recipient) #invalid recipient
+    let (token_owner) = _owners.read(token_id)
+    assert token_owner = 0 #already minted
+
+    let (current_balance) = _balances.read(owner=recipient)
+    let (new_balance, _: Uint256) = uint256_add(current_balance, Uint256(0,1))
```

Comment:
is the `Uint256` type cast irrelevant here since it's typing the ignored `_` element, and the `new_balance` element is inferred from the function's return type (_i think_)